neutrons/SNAPRed
# Ewm5730 diagnostic workspace crash

## Description of work

This pr covers 2 bug fixes:
1. The GroupDiffraction algo trying to join table workspaces with mismatching columns causing it to crash.
2. ~Reduction Workflow unable to proceed because the runnumber submitted to the ReductionRequest object is not a str~  This will be taken care of in pr #393 

## To test

1. Refer to the story for the diffraction calibration input required
2. ~Submit a reduction via the workflow, observe no validation exception occurs.  Ideally reduction should complete granted you have a complete state folder.~

## Link to EWM item


[EWM#5730](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=5730)
